core/src/main/java/org/mule/runtime/core/internal/util/message/MessageUtils.java

```diff
@@ -517,31 +477,14 @@ private static Message toMessage(Result<?, ?> result, DataType dataType, Object
 
   /**
    * Decorates input value.
-   * 
-   * @param v                         value to be decorated
-   * @param eventCorrelationId        the correlationId of the context involved
-   * @param componentDecoratorFactory the component decorator factory
-   * 
+   *
+   * @param v value to be decorated
+   *
    * @return decorated value
    */
-  public static Object decorateInput(Object v, String eventCorrelationId,
-                                     CursorComponentDecoratorFactory componentDecoratorFactory) {
-    if (v instanceof byte[]) {
-      componentDecoratorFactory.computeInputByteCount((byte[]) v);
-      return v;
-    } else if (v instanceof CursorStream) {
-      return componentDecoratorFactory.decorateInput((InputStream) new UnclosableCursorStream((CursorStream) v),
-                                                     eventCorrelationId);
-    } else if (v instanceof InputStream) {
-      return componentDecoratorFactory.decorateInput((InputStream) v, eventCorrelationId);
-    } else if (v instanceof Collection) {
-      return componentDecoratorFactory.decorateInput((Collection) v, eventCorrelationId);
-    } else if (v instanceof Iterator) {
-      return componentDecoratorFactory.decorateInput((Iterator) v, eventCorrelationId);
-    } else if (v instanceof CursorStreamProvider) {
-      return new InputDecoratedCursorStreamProvider((CursorStreamProvider) v,
-                                                    componentDecoratorFactory,
-                                                    eventCorrelationId);
+  public static Object decorateInput(Object v) {
+    if (v instanceof CursorStream) {
+      return new UnclosableCursorStream((CursorStream) v);
```

Comment:
Had to leave this part because of #11209.

Reply:
if this is still needed then change signature to be more clear. It should be called asUncloseableStream and should explicitly receive an InputStream.

Reply:
I considered making it receive an InputStream (or even CursorStream) but, if we do that, then we would have to place the `instanceof` check on many different places (basically all places that are now calling this method). However they may be reduced if the wrapping is not necessary on `DefaultObjectBuilder` and `ParameterGroupObjectBuilder`.

I could even move this inside `MethodArgumentResolverDelegate`.

Reply:
exactly my point, this method should only be invoked when we know that the value may be a stream.

Reply:
This method no longer exists